python/glow/wgr/logistic_ridge_regression.py

```diff
@@ -0,0 +1,325 @@
+    provided list of ridge alpha parameters. The optimal ridge alpha value is chosen for each label by minimizing the
+    average out of fold log_loss scores.
+    """
+    def __init__(
+        self,
+        reduced_block_df: DataFrame,
+        label_df: pd.DataFrame,
+        sample_blocks: Dict[str, List[str]],
+        cov_df: pd.DataFrame = pd.DataFrame({}),
+        add_intercept: bool = True,
+        alphas: NDArray[(Any, ), Float] = np.array([])) -> None:
+        """
+        Args:
+            reduced_block_df : Spark DataFrame representing the reduced block matrix generated by
+                RidgeReduction
+            label_df : Pandas DataFrame containing the target labels used in fitting the ridge models
+            sample_blocks : Dict containing a mapping of sample_block ID to a list of corresponding sample IDs
+            cov_df : Pandas DataFrame containing covariates to be included in every model in the stacking
+                ensemble (optional).
+            add_intercept: If True, an intercept column (all ones) will be added to the covariates
+                (as the first column)
+            ridge_reduced: RidgeReduction object containing level 0 reduction data
+            alphas : array_like of alpha values used in the ridge regression (optional).
+        """
+        self.reduced_block_df = reduced_block_df
+        self.sample_blocks = sample_blocks
+        self.set_label_df(label_df)
+        self.set_cov_df(cov_df, add_intercept)
+        self.set_alphas(alphas)
+        self.model_df = None
+        self.cv_df = None
+        self.y_hat_df = None
+
+    @classmethod
+    def from_ridge_reduction(cls,
+                             ridge_reduced: RidgeReduction,
+                             alphas: NDArray[(Any, ), Float] = np.array([])):
+        """
+        Initializes an instance of LogsiticRidgeRegression using a RidgeReduction object
+
+        Args:
+            ridge_reduced : A RidgeReduction instance based on which the LogisticRidgeRegression instance must be made
+            alphas : array_like of alpha values used in the logistic ridge regression (optional).
+        """
+        obj = cls.__new__(cls)
+        obj.reduced_block_df = ridge_reduced.reduced_block_df
+        obj.sample_blocks = ridge_reduced.sample_blocks
+        obj._label_df = ridge_reduced.get_label_df()
+        obj._cov_df = ridge_reduced.get_cov_df()
+        obj._std_cov_df = ridge_reduced._std_cov_df
+        obj.set_alphas(alphas)
+        obj.model_df = None
+        obj.cv_df = None
+        obj.y_hat_df = None
+        return obj
+
+    def __getstate__(self):
+        # Copy the object's state from self.__dict__ which contains
+        state = self.__dict__.copy()
+        # Remove the unpicklable entries.
+        del state['reduced_block_df'], state['model_df'], state['cv_df']
+        return state
+
+    def __setstate__(self, state):
```

Comment:
Do we actually need a custom `__setstate__` implementation? What happens if this is omitted?